### PR TITLE
[flatbuffers] update to 2.0.8

### PR DESCRIPTION
**Pull request by chausner**
Specify library name and version:  **flatbuffers/2.0.8**

This updates flatbuffers to the latest version 2.0.8.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/flatbuffers/all/conanfile.py [hunk 1]
```diff
@@ -132,7 +132,10 @@ def _module_path(self):
     def package_info(self):
         self.cpp_info.set_property("cmake_find_mode", "both")
         self.cpp_info.set_property("cmake_module_file_name", "FlatBuffers")
-        self.cpp_info.set_property("cmake_file_name", "Flatbuffers")
+        if Version(self.version) < "2.0.7":
+            self.cpp_info.set_property("cmake_file_name", "Flatbuffers")
+        else:
+            self.cpp_info.set_property("cmake_file_name", "FlatBuffers")
```
Comment (chausner) [inline recipes/flatbuffers/all/conanfile.py:138]: See https://github.com/google/flatbuffers/pull/7378.

Comment (SpaceIm) [reply recipes/flatbuffers/all/conanfile.py:138]: prefer https://github.com/google/flatbuffers/pull/7502

Comment (chausner) [reply recipes/flatbuffers/all/conanfile.py]: I changed it in d4c03f3a3267b27df147488e0878fab136f76c06.

File: recipes/flatbuffers/all/conanfile.py [hunk 2]
```diff
@@ -43,6 +43,10 @@ def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
+    def build_requirements(self):
+        if Version(self.version) >= "2.0.7":
+            self.tool_requires("cmake/3.24.0")
```
Comment (chausner) [inline recipes/flatbuffers/all/conanfile.py:48]: CMake 3.16 is now the minimum supported version number. They have a legacy CMakeLists.txt for older CMake versions which is unmaintained, however, and does not work.

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**